cloud/pkg/synccontroller/synccontroller.go

```diff
@@ -223,6 +244,59 @@ func (sctl *SyncController) manageObjectSync(syncs []*v1alpha1.ObjectSync) {
 	}
 }
 
+// 	when CloudCore starts, it will delete outdated syncs, then triggered by node delete events.
+func (sctl *SyncController) DeleteOutdatedObjectSync() {
+	sctl.deleteObjectSyncs()
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop synccontroller DeleteOutdatedObjectSync loop")
+			return
+		case e := <-sctl.nodeManager.Events():
+			node, ok := e.Object.(*v1.Node)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", node)
+				continue
+			}
+			switch e.Type {
+			case watch.Deleted:
+				sctl.deleteObjectSyncs()
+			}
+		}
+	}
+}
+
+func (sctl *SyncController) deleteObjectSyncs() {
+	syncs, err := sctl.objectSyncLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("Failed to list all the ObjectSyncs: %v", err)
+	}
+	for _, sync := range syncs {
+		nodeName := getNodeName(sync.Name)
+		isGarbage, err := sctl.checkObjectSync(sync)
+		if err != nil {
+			klog.Errorf("Fail to check ObjectSync outdated, %s", err)
+		}
+		if isGarbage {
+			klog.Infof("ObjectSync %v will be deleted because node %v unregisters  ", sync.Name, nodeName)
+			err = sctl.crdClient.ReliablesyncsV1alpha1().ObjectSyncs(sync.Namespace).Delete(sync.Name, metav1.NewDeleteOptions(0))
+			if err != nil {
+				klog.Errorf("Fail to delete objectSync %v of EdgeNode %v which is offline", sync.Name, nodeName)
+			}
+		}
+	}
+}
+
+// checkObjectSync checks whether objectSync is outdated
+func (sctl *SyncController) checkObjectSync(sync *v1alpha1.ObjectSync) (bool, error) {
+	nodeName := getNodeName(sync.Name)
+	_, err := sctl.nodeLister.Get(nodeName)
```

Comment:
@daixiang0 I read the underlying implementation of nodeLister.Get, I found err here is always nil even if it maybe has wrong. the err has two value, nil or NotFound. (in line206 /client-go@v0.17.1/tools/cache/store.go)

Reply:
Let's keep sync with it :)